Remove "redundant null check due to previous dereference" found in CodeQL scan

esabol opened the pull request:
This PR fixes an "issue" reported by the CodeQL scan of the gearmand source code. The null check on [line 179 of libtest/server.cc](https://github.com/gearman/gearmand/blob/8e80a0ded3495496c7037a50eab7fc680c29b2e9/libtest/server.cc#L177-L179) is redundant because it has already been dereferenced on line 177 in the condition of the `for` loop.
![Screen Shot 2023-11-12 at 6 33 05 PM](https://github.com/gearman/gearmand/assets/22986767/ba6b22f6-cf68-48a9-a460-ce7888377917)

I suppose the `if (argv) { ... }` around the `for` loop could also be removed since the new condition of the `for` loop will also test this, but it doesn't do any harm and arguably improves clarity. I'd be willing to remove it though if you think it should be removed, @SpamapS.

SpamapS reviewed (approved):
Wow sometimes maintaining gearmand feels like working on an archeological dig.

Agree with your assessment. Remove that extra check. It might even get optimized out in some cases.

Anyway, I looked at some of the calls to this and funny enough the only one I could find that uses the argv's was my own addition of --round-robin mode.

esabol commented:
@SpamapS wrote:
> Agree with your assessment. Remove that extra check. It might even get optimized out in some cases.

Yeah, I think it would be with any good compiler. OK, I've removed the `if (argv) { ... }` as well. Squashed and rebased.
